roles/core/dns_server/tasks/main.yml

```diff
@@ -43,25 +64,9 @@
   tags:
     - template
 
-- name: Template >> /var/named/forward
-  template:
-    src: forward.j2
-    dest: /var/named/forward
-    owner: root
-    group: root
-    mode: 0644
-  notify: Restart dns services
-  tags:
-    - template
-
-- name: Template >> /var/named/reverse
-  template:
-    src: reverse.j2
-    dest: /var/named/reverse
-    owner: root
-    group: root
-    mode: 0644
-  notify: Restart dns services
+- name: Configure Master DNS
+  include_tasks: master.yml
+  when: "dns_master in ansible_all_ipv4_addresses"
```

Comment:
After spending some time computing on this, I don't like the usage of ansible_all_ipv4_addresses here. I will explain in a next comment in detail.

Reply:
Done